client/state/terms/selectors.js

```diff
@@ -26,6 +27,28 @@ export function isRequestingTermsForQuery( state, siteId, taxonomy, query ) {
 }
 
 /**
+ * Returns true if currently requesting terms for a query, excluding all known
+ * queried pages, or false otherwise.
+ *
+ * @param  {Object}  state    Global state tree
+ * @param  {Number}  siteId   Site ID
+ * @param  {String}  taxonomy Taxonomy slug
+ * @param  {Object}  query    Terms query object
+ * @return {Boolean}           Terms for the query
+ */
+export function isRequestingTermsForQueryIgnoringPage( state, siteId, taxonomy, query ) {
```

Comment:
You should use `TermQueryManager#getItemsIgnoringPage` to implement this ([see equivalent posts implementation](https://github.com/Automattic/wp-calypso/blob/257ec0c69d25d750c02af22eec409676f685daf5/client/state/posts/selectors.js#L171-L194)). You can drop `range` dependency after too.